optimiser/optimisation_problem.py

```diff
@@ -0,0 +1,220 @@
+import pulp as pl
+from pulp import LpVariable
+
+
+def main():
+    # Define the problem
+    problem = pl.LpProblem("BatteryOptimisation", pl.LpMinimize)
+
+    # Time slices
+    T = range(1, 25)
+
+    TIMESLICE_SIZE = 0.5
+
+    # Renewable Generation Constants
+    MAX_RENEWABLE_GENERATION = 10  # kWh
```

Comment:
Max generation should be a variable and change with time. Add to a separate issue

Reply:
Raised issue with https://github.com/davidwickh/Home-Battery-Optimiser/issues/8